server/src/main/java/org/elasticsearch/common/time/EpochSecondsDateFormatter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.time;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAccessor;
+import java.time.temporal.TemporalField;
+import java.util.Map;
+
+public class EpochSecondsDateFormatter implements DateFormatter {
+
+    public static DateFormatter INSTANCE = new EpochSecondsDateFormatter();
+
+    private EpochSecondsDateFormatter() {}
+
+    @Override
+    public TemporalAccessor parse(String input) {
+        try {
+            if (input.contains(".")) {
+                String[] inputs = input.split("\\.", 2);
+                Long seconds = Long.valueOf(inputs[0]);
+                if (inputs[1].length() == 0) {
+                    // this is BWC compatible to joda time, nothing after the dot is allowed
+                    return Instant.ofEpochSecond(seconds, 0).atZone(ZoneOffset.UTC);
+                }
+                if (inputs[1].length() > 9) {
+                    throw new DateTimeParseException("too much granularity after dot [" + input + "]", input, 0);
+                }
+                Long nanos = new BigDecimal(inputs[1]).movePointRight(9 - inputs[1].length()).longValueExact();
+                return Instant.ofEpochSecond(seconds, nanos).atZone(ZoneOffset.UTC);
+            } else {
+                return Instant.ofEpochSecond(Long.valueOf(input)).atZone(ZoneOffset.UTC);
+            }
+        } catch (NumberFormatException e) {
+            throw new DateTimeParseException("invalid number", input, 0, e);
+        }
+    }
+
+    @Override
+    public DateFormatter withZone(ZoneId zoneId) {
+        return this;
```

Comment:
This seems trappy...we should at least have a comment why this is ok, but it seems like we could drop timezones?

Reply:
the java-time branch will create new objects with a timezone for all the date formatters (same needs to be done for the locale). I will create a separate PR for that after this one.